etl/steps/data/garden/faostat/2022-05-17/shared.py

```diff
@@ -980,49 +980,51 @@ def prepare_wide_table(data: pd.DataFrame, dataset_title: str) -> catalog.Table:
     # Note: We include area_code in the index for completeness, but by construction country-year should not have
     # duplicates.
     # Note: `pivot` operation is usually faster on categorical columns
-    data_pivot = data.pivot(index=["area_code", "country", "year"], columns=["variable_name"],
-                            values=["value", "unit", "unit_short_name", "unit_factor", "variable_display_name",
-                                    "variable_description"])
-
-    # For convenience, create a dictionary for each zeroth-level multi-index.
-    data_wide = {pivot_column: data_pivot[pivot_column] for pivot_column in data_pivot.columns.levels[0]}
-
+    log.info("prepare_wide_table.pivot", shape=data.shape)
     # Create a wide table with just the data values.
-    wide_table = catalog.Table(data_wide["value"]).copy()
+    wide_table = catalog.Table(
+        data.pivot(index=["area_code", "country", "year"], columns=["variable_name"], values="value")
+    )
 
-    # Add metadata to each new variable in the wide data table.
+    ### Add metadata to each new variable in the wide data table.
+    log.info("prepare_wide_table.adding_metadata", shape=wide_table.shape)
+
+    # Add variable name.
     for column in wide_table.columns:
-        # Add variable name.
         wide_table[column].metadata.title = column
 
-        # Add variable unit (long name).
-        variable_unit = data_wide["unit"][column].dropna().unique()
-        assert len(variable_unit) == 1
-        wide_table[column].metadata.unit = variable_unit[0]
+    # Add variable unit (long name).
+    m = _variable_name_map(data, "unit")
```

Comment:
I'd prefer to call `m` something more meaningful, like `variable_name_mapping`, but it's up to you.